orchagent/routeorch.cpp

```diff
@@ -373,187 +376,318 @@ void RouteOrch::doTask(Consumer& consumer)
     auto it = consumer.m_toSync.begin();
     while (it != consumer.m_toSync.end())
     {
-        KeyOpFieldsValuesTuple t = it->second;
-
-        string key = kfvKey(t);
-        string op = kfvOp(t);
-
-        /* Get notification from application */
-        /* resync application:
-         * When routeorch receives 'resync' message, it marks all current
-         * routes as dirty and waits for 'resync complete' message. For all
-         * newly received routes, if they match current dirty routes, it unmarks
-         * them dirty. After receiving 'resync complete' message, it creates all
-         * newly added routes and removes all dirty routes.
-         */
-        if (key == "resync")
+        // Route bulk results will be stored in a map
+        std::map<
+                std::pair<
+                        std::string,            // Key
+                        std::string             // Op
+                >,
+                RouteBulkContext
+        >                                       toBulk;
+
+        // Add or remove routes with a route bulker
+        while (it != consumer.m_toSync.end())
         {
-            if (op == "SET")
+            KeyOpFieldsValuesTuple t = it->second;
+
+            string key = kfvKey(t);
+            string op = kfvOp(t);
+
+            auto rc = toBulk.emplace(std::piecewise_construct,
+                    std::forward_as_tuple(key, op),
+                    std::forward_as_tuple());
+
+            bool inserted = rc.second;
+            auto& ctx = rc.first->second;
+            if (!inserted)
             {
-                /* Mark all current routes as dirty (DEL) in consumer.m_toSync map */
-                SWSS_LOG_NOTICE("Start resync routes\n");
-                for (auto j : m_syncdRoutes)
-                {
-                    string vrf;
+                ctx.clear();
+            }
 
-                    if (j.first != gVirtualRouterId)
+            /* Get notification from application */
+            /* resync application:
+             * When routeorch receives 'resync' message, it marks all current
+             * routes as dirty and waits for 'resync complete' message. For all
+             * newly received routes, if they match current dirty routes, it unmarks
+             * them dirty. After receiving 'resync complete' message, it creates all
+             * newly added routes and removes all dirty routes.
+             */
```

Comment:
Actually this is old code. Since the nature of this PR, I need to minimize the changes in favor or easy reviewing. We could focus on refactor later.

---
In reply to: [408379175](https://github.com/Azure/sonic-swss/pull/1238#discussion_r408379175) [](ancestors = 408379175)

Reply:
Ah, okay! I wasn't aware of the history here. In that case, I agree with you that it's probably best to save the refactoring for later. :)